### PR TITLE
chore: add docs for retry

### PR DESCRIPTION
**Pull request by kohlisid**

### CODE REVIEW

**Comment by whynowy** [comment]
Add it to mkdocs.yaml

**Comment by whynowy** [comment]
Under `sinks`, after `fallback sink`

**Review by whynowy — APPROVED** [review]

File: docs/user-guide/sinks/retry-strategy.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+`retryStrategy` is optional, and can be added to the Sink spec configurations where retry logic is necessary. 
+
+```yaml
+sink:
+  retryStrategy:
+      # Optional
+      backoff:
+        duration: 1s # Optional, defaults 1ms
+        steps: 3 # Optional, number of retries (including the 1st try)
```
Comment (vigith) [inline docs/user-guide/sinks/retry-strategy.md:20]: what is the default for `steps`? you have written the default for other two.

File: docs/user-guide/sinks/retry-strategy.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+- BackOff - Defines the timing for retries, including the interval and the maximum attempts.
+  - duration: the time interval to wait before retry attempts
+  - steps: the number of times to try the sink write operation
+- OnFailure - Specifies the action to be undertaken if number of retries are exhausted
+  - retry: continue with the retry logic again
+  - fallback: write the leftover messages to a [fallback](https://numaflow.numaproj.io/user-guide/sinks/fallback/) sink
+  - drop: any messages left to be processed are dropped
+
+Note: If no custom fields are defined for retryStrategy then the default values are used.
```
Comment (vigith) [inline docs/user-guide/sinks/retry-strategy.md:33]: merge it with the above.

File: docs/user-guide/sinks/retry-strategy.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      onFailure: "fallback"
+    udsink:
+      container:
+        image: my-sink-image
+    fallback:
+      udsink:
+        container:
+          image: my-fallback-sink
+```
+#### Explanation
```
Comment (vigith) [inline docs/user-guide/sinks/retry-strategy.md:63]:
```suggestion
### Explanation
```

**Pull request merged**